docs/reports/tootsie-retro.md

```diff
@@ -0,0 +1,416 @@
+or the [WandB report](https://wandb.ai/stanford-mercury/marin/reports/Tootsie-8B---VmlldzoxMTY3MzU3OA)
+
+
+## The Origin of "Tootsie"
+
+A core premise of the Tootsie 8B run was that we didn't fully know what the best recipe was, so we decided to just start
+training something with the best information we had.
+As new information (data, techniques, etc.) became available, we'd fold it into the training run and just keep chugging.
+
+The name "Tootsie" comes from [Tootsie Rolls, which use a "graining" process](https://en.wikipedia.org/wiki/Tootsie_Roll) whereby a portion of the previous
+day's batch is folded into the next, seeding the crystallization process or some such. This is admittedly a bit of a
+tortured metaphor, but the idea was that we would expect to change the data recipe as training progressed.
+(As it would turn out, dear reader, we would often change more than the data...)
+
+
+## Model Basics
+
+### Model Size
+
+We decided to build an ~7-8B class model mostly out of pragmatism: we initially only had reserved capacity
+to train a model of that size for long enough.
+
+### Architecture
+
+We settled on the "Llama architecture" for the usual reasons: it has been shown to work well, no one ever got fired
+for buying IBM, etc.
+
+We used Levanter's implementation of the Llama architecture.
+We trained with a sequence length of 4096 tokens per sample.
+We used JAX's TPU Splash Attention kernel.
+
+We used mixed float32/bfloat16 precision, with parameters and optimizer states in float32 and compute in bfloat16
+(except for the final softmax over the vocabulary.)
+We used the AdamW optimizer.
+
+### Tokenizer
+
+In Marin, we also standardized on the Llama 3 tokenizer, after [an experiment](https://github.com/marin-community/marin/issues/524) showing it to be superior
+to both Llama 2 and NeoX in terms of bits-per-byte (bpb).
+
+### Batch Schedule
+
+We used a varying batch schedule, with between 1024 * 4096 = 4Mi tokens, 3072 * 4096 = 12 Mi tokens and 4096 * 4096 = 16 Mi tokens.
+(As with most things Tootsie, we did not initially plan for this schedule.)
+(Note, we use `Mi` to mean Mebi-, not mega-.)
+
+### Checkpointing Policy
+
+We saved permanent full checkpoints every 20,000 steps (which, due to the varying batch schedule, could be a varying number of tokens.)
+To save on egress bandwidth charges, we have not made these available. However, if they are of interest we should be
+able to share them.
+
+("Temporary" checkpoints were saved much more frequently.)
+
+# Training Phases
+
+Retrospectively, we can partition the 8b run into several different phases. We have given them animal names as monikers.
+
+* *DCLM WSD-S Phase*: In the first phase, we used the "DCLM mix" and [WSD-S](https://arxiv.org/abs/2410.05192)  for about 2.8T tokens. We used 2x TPU v5e-256 coordinated with multislice for this.
```

Comment:
Maybe want to say why we were excited about WSD-S despite it being a bit non-standard...and foreshadow our conclusions later